pkg/controllers/jobset_controller.go

```diff
@@ -100,22 +131,22 @@ func (r *JobSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	log.V(2).Info("Reconciling JobSet")
 
 	// Get Jobs owned by JobSet.
-	ownedJobs, err := r.getChildJobs(ctx, &js)
+	ownedJobs, err := r.getChildJobs(ctx, js)
 	if err != nil {
 		log.Error(err, "getting jobs owned by jobset")
 		return ctrl.Result{}, err
 	}
 
-	// Calculate JobsReady and update statuses for each ReplicatedJob
-	status := r.calculateReplicatedJobStatuses(ctx, &js, ownedJobs)
-	if err := r.updateReplicatedJobsStatuses(ctx, &js, ownedJobs, status); err != nil {
-		log.Error(err, "updating replicated jobs statuses")
-		return ctrl.Result{}, err
+	// Calculate JobsReady and update statuses for each ReplicatedJob.
+	// If status ReplicatedJobsStatus has changed, update the JobSet status.
+	rjobStatuses := r.calculateReplicatedJobStatuses(ctx, js, ownedJobs)
+	if !replicatedJobStatusesEqual(js.Status.ReplicatedJobsStatus, rjobStatuses) {
```

Comment:
This check probably fits better inside `updateReplicatedJobsStatuses`

Reply:
Done